src/linear_algebra/special_linear_group.lean

```diff
@@ -0,0 +1,133 @@
+/-
+  Copyright (c) 2020 Anne Baanen. All rights reserved.
+  Released under Apache 2.0 license as described in the file LICENSE.
+  Author: Anne Baanen.
+
+  The Special Linear group $SL(n, R)$
+-/
+import linear_algebra.basic
+import linear_algebra.matrix
+import linear_algebra.nonsingular_inverse
+import tactic.norm_cast
+
+/-!
+# The Special Linear group $SL(n, R)$
+
+This file defines the elements of the Special Linear group `special_linear_group n R`,
+also written `SL(n, R)` or `SLₙ(R)`, consisting of all `n` by `n` `R`-matrices with
+determinant `1`.  In addition, we define the group structure on `special_linear_group n R`
+and the embedding into the general linear group `general_linear_group R (n → R)`
+(i.e. `GL(n, R)` or `GLₙ(R)`).
+
+## Main definitions
+
+ * `matrix.special_linear_group` is the set of matrices with determinant 1
+ * `matrix.special_linear_group.group` gives the group structure (under multiplication)
+ * `matrix.special_linear_group.to_GL` is the embedding `SLₙ(R) → GLₙ(R)`
+
+## Implementation notes
+The inverse operation in the `special_linear_group` is defined to be the adjugate
+matrix, so that `special_linear_group n R` has a group structure for all `comm_ring R`.
+
+We define the elements of `special_linear_group` to be matrices, since we need to
+compute their determinant. This is in contrast with `general_linear_group R M`,
+which consists of invertible `R`-linear maps on `M`.
```

Comment:
Should we put this `special_linear_group` in the `matrix` namespace? We could also put another `general_linear_group` in the `matrix` namespace. Idem dito for orthogonal groups etc.
Then we can reserve `special_linear_group R V` for the group of determinant 1 linear endomorphisms.

Reply:
These definitions already live in the `matrix` namespace :)

Reply:
Aah, sorry. I missed that.